python/pyarrow/parquet.py

```diff
@@ -1244,7 +1255,7 @@ def __init__(self, path_or_paths, filesystem=None, schema=None,
                  metadata=None, split_row_groups=False, validate_schema=True,
                  filters=None, metadata_nthreads=1, read_dictionary=None,
                  memory_map=False, buffer_size=0, partitioning="hive",
-                 use_legacy_dataset=True):
+                 use_legacy_dataset=True, pre_buffer=True):
```

Comment:
Should we maybe raise an error here if the user sets it to False? (as we simply ignore it)

Reply:
I don't think so as False is equivalent to the "not supported" behavior. I could set the default value to 'None' and let each individual ParquetDataset implementation handle it (and error on True in this case).

Reply:
Ah, yes, since `pre_buffer=False` means to not do anything, it's indeed fine to "ignore" it. 
As you say, we could do some extra work to let it error on `pre_buffer=True` (for the legacy implementation), but since it's clearly documented to be only for the new version, I think it's fine to leave it as is.